frontend/src/host_orchestrator/orchestrator/createcvdaction.go

```diff
@@ -337,3 +356,57 @@ func validateRequest(r *apiv1.CreateCVDRequest) error {
 	}
 	return nil
 }
+
+// See https://pkg.go.dev/io/ioutil@go1.13.15#TempFile
+func createTempFile(pattern string, data []byte, mode os.FileMode) (*os.File, error) {
+	file, err := ioutil.TempFile("", pattern)
+	if err != nil {
+		return nil, err
+	}
+	if err := file.Chmod(mode); err != nil {
+		return nil, err
+	}
+	_, err = file.Write(data)
+	if closeErr := file.Close(); closeErr != nil && err == nil {
+		err = closeErr
+	}
+	if err != nil {
+		return nil, err
+	}
+	return file, nil
+}
+
+func createCredsFile(ctx cvd.CVDExecContext) (string, error) {
+	name, err := tempFilename("cvdload*.creds")
+	if err != nil {
+		return "", err
+	}
+	if err := cvd.Exec(ctx, "touch", name); err != nil {
+		return "", err
+	}
+	if err := cvd.Exec(ctx, "chgrp", "cvdnetwork", name); err != nil {
+		return "", err
+	}
+	if err := cvd.Exec(ctx, "chmod", "0620", name); err != nil {
+		return "", err
+	}
```

Comment:
Why is this done with three commands instead of
```
os.WriteFile(name, credentials, 0620)
```
?

Reply:
Using os.WriteFile(), `_cutf-operator` will be the owner of the file. Created the file this way to have the cvd user: ` _cvd-executor` being the owner instead.

Reply:
A comment saying just that would be nice here.

Reply:
very true, done.